### ISSUE TITLE
helm3  - can't override the ConfigMap

### ISSUE THREAD

**Issue by cynial** [helm/helm#6475]
helm version: 
v3.0.0-beta.3

kubectl version:
1.10.5

---

Upgrade haproxy-ingress 

`helm upgrade haproxy-ingress haproxy-ingress`

```
# values.yaml

controller:
  config:
    config-defaults: |
      option forwardfor header X-Real-IP
```

Check the configmap 

`kubectl describe configmap haproxy-ingress-controller`

```
Data
====
config-defaults:
----
option forwardfor header X-Real-IP
```
 ---

Change the configmap to: 

```
# values.yaml

controller:
  config:
    config-defaults: |
      option forwardfor header X-Real-IP
      option redispath
```

`helm upgrade haproxy-ingress haproxy-ingress`
`kubectl describe configmap haproxy-ingress-controller`

```
Data
====
config-defaults:
----
option forwardfor header X-Real-IP
```

Can't change the value of config-defaults.

---

It‘s working fine on helm2, but not helm3, has anyone encountered this issue?

**Comment by thomastaylor312** [comment]
@cynial Could you try passing `-f values.yaml` on your upgrade and see if that works? I think I may have seen this before

**Comment by karuppiah7890** [comment]
I'm guessing this is related to https://github.com/helm/helm/issues/6358 where the chart got installed with default values.yaml and the release has computed values instead of nil. And then the chart default values.yaml is changed and upgraded with no `--values` or `-f` and this time, since there are no values yamls passed and release has values, that is used and overrides the chart's default values.yaml and therefore the final values is the one that comes with the installation. 

`$ helm get values <release-name> --revision <revision-number> --namespace <namespace>` 

Using the above command and checking values for different revisions should show that that's the issue

 I also raised a PR for the same https://github.com/helm/helm/pull/6430 . Hoping to get it reviewed and merged 😄

**Comment by thomastaylor312** [comment]
Fixed in #6430
